### DISCUSSION TITLE
Disconnect PnP Online with connections parameter

### DISCUSSION THREAD

**Discussion by pratheep-r** [pnp/powershell#3239] [Q&A] [answered]
As a part of the script, within "foreach" loop I am trying to disconnect in "finally" block. However getting the attached error

Code in nutshell
 
`$Cred = Get-Credential
try {
    $Connection = Connect-PnPOnline -ReturnConnection -url "https://tenant/sites/EmergencyResponse" -Credentials $Cred
    # Do something
}
catch {
    write-host -f Red "`tError:" $_.Exception.Message
}
finally {
    If ($null -ne $Connection) {
        Disconnect-PnPOnline -Connection $Connection
    }
}`

Please advise/help. 
![2023-07-03 17_01_38-● Test ps1 - DisconnectPnP](https://github.com/pnp/powershell/assets/56027088/d44e05eb-5636-413a-bf1f-04f4b653007d)

**Comment by AndersRask** [comment, marked as answer]
As the error say, connection parameter does not exist.

It was removed several versions ago. I general you should not disconnect
connections, just make new ones with new sites.

man. 3. jul. 2023 18.11 skrev Pra-R ***@***.***>:

**Reply by pratheep-r** [reply]
Ah okay, thanks. Are we not running the risk of leaving it connected?

**Comment by AndersRask** [comment]
No, it's actually recommended not to (read on the description for the
cmdlet) since it's the credential that is cached

man. 3. jul. 2023 18.27 skrev Pra-R ***@***.***>:
